build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/ui/wizards/MBSWizardHandler.java

```diff
@@ -611,20 +610,15 @@ private void setProjectDescription(IProject project, boolean defaults, boolean o
 		cfgs = CfgHolder.unique(cfgs);
 		cfgs = CfgHolder.reorder(cfgs);
 
-		ICConfigurationDescription cfgDebug = null;
-		ICConfigurationDescription cfgFirst = null;
 		subMonitor.worked(1);
 
+		IConfiguration active = null;
 		SubMonitor cfgMonitor = SubMonitor.convert(subMonitor.split(1), cfgs.length);
 		for (CfgHolder cfg : cfgs) {
 			cf = (Configuration) cfg.getConfiguration();
-			String id = ManagedBuildManager.calculateChildId(cf.getId(), null);
-			Configuration config = new Configuration(mProj, cf, id, false, true);
-			CConfigurationData data = config.getConfigurationData();
-			ICConfigurationDescription cfgDes = des.createConfiguration(ManagedBuildManager.CFG_DATA_PROVIDER_ID, data);
-			config.setConfigurationDescription(cfgDes);
-			config.exportArtifactInfo();
-
+			IConfiguration config = ManagedBuildManager.createConfigurationForProject(des, mProj, cf,
+					ManagedBuildManager.CFG_DATA_PROVIDER_ID);
+			// TODO check 			config.exportArtifactInfo();
```

Comment:
Looks like this call to exportArtifactInfo() was moved to createConfigurationForProject()

What is TODO here?

Reply:
Forgot to delete that. It is a reminder to check whether the refactored code should call 'config.exportArtifactInfo();'.